drivers/flash/nrf_qspi_nor.c

```diff
@@ -491,15 +491,36 @@ static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 		return -EINVAL;
 	}
 
-	/* read size must be non-zero multiple of 4 bytes */
-	if (((size % 4U) != 0) || (size == 0)) {
-		return -EINVAL;
-	}
 	/* address must be 4-byte aligned */
 	if ((addr % 4U) != 0) {
 		return -EINVAL;
 	}
 
+#if IS_ENABLED(CONFIG_NORDIC_QSPI_NOR_FLASH_ONE_BYTE_RW_ACCESS)
+	void *tmp_dest = dest;
+	size_t var_size = size;
+
+	if ((size < sizeof(u32_t)) && (size != 0)) {
```

Comment:
Should there also be a check for max size which is Kconfigurable to ensure that the function won't use infinite stack?

Reply:
I think it's reasonable for a hack like this to say it only works for sizes less than, say, 8 bytes, which would require a 16-byte buffer on the stack.  Misaligned/badly-sized transfers larger than 8 bytes can be rejected.

Reply:
Okei, I'll add it as a Kconfig. I'll also see if I can get it working to only allocate the array on the stack if < 4 or not in RAM.